cpp/CMakeLists.txt

```diff
@@ -334,6 +334,14 @@ if(RAFT_COMPILE_LIBRARY)
     src/matrix/detail/select_k_float_int32.cu
     src/matrix/detail/select_k_half_int64_t.cu
     src/matrix/detail/select_k_half_uint32_t.cu
+    src/sparse/matrix/detail/select_k_half_uint32_t.cu
```

Comment:
We're removing libraft.so in a future version, so I think we probably want to avoid adding these instantiations.